From 2ffb9c8ca3b4a5ae53cb1ea55a6dcd0db0816ea5 Mon Sep 17 00:00:00 2001
From: 28 <Rajoduo@yahoo.com>
Date: Thu, 2 Feb 2017 20:47:05 +0100
Subject: [PATCH] CLJ-1868 - Output simpler dependency rel paths when compiling
 with Closure libs

Until now compiler did not parse input paths for Closure dependencies correctly
and as a result the output paths contained full absolute path as its part.
That path contained charachters like : and ! that were ok on unix machines
but caused problem on windows machines. With this fix the compiler will
output correct rel paths.
---
 src/main/clojure/cljs/closure.clj | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/main/clojure/cljs/closure.clj b/src/main/clojure/cljs/closure.clj
index ca97f8d..e5248a4 100644
--- a/src/main/clojure/cljs/closure.clj
+++ b/src/main/clojure/cljs/closure.clj
@@ -1481,10 +1481,7 @@
     (if (.endsWith lib-path ".js")
       (util/get-name url)
       (let [path (util/path url)]
-        (string/replace
-          path
-          (str (io/file (System/getProperty "user.dir") lib-path) File/separator)
-          "")))))
+        (subs path (+ (.lastIndexOf path lib-path) (.length lib-path)))))))
 
 (defn ^String rel-output-path
   "Given an IJavaScript which is either in memory, in a jar file,
-- 
2.9.3

